feat: new Rating Indicator component

Betrozov opened the pull request:
## Related Issue
Closes https://github.com/SAP/fundamental-styles/issues/1431

## Description
Introducing new Rating Indicator component

## Screenshots
### After:
<img width="304" alt="Screenshot 2020-08-27 at 01 21 06" src="https://user-images.githubusercontent.com/8318592/91362765-ce2fb380-e803-11ea-95e7-0acb62089050.png">
<img width="924" alt="Screenshot 2020-08-27 at 01 21 27" src="https://user-images.githubusercontent.com/8318592/91362782-d851b200-e803-11ea-82bc-91405d0b35e2.png">

#### Please check whether the PR fulfills the following requirements

- [ x ] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-styles/wiki/PR-Review-Checklist

InnaAtanasova commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,119 @@
+    "condensed": ("fontSize": 1rem, "spacingBetweenIcons": 0.125rem),
+    "xs": ("fontSize": 0.75rem, "spacingBetweenIcons": 1px),
+    "s": ("fontSize": 1rem, "spacingBetweenIcons": 0.125rem),
+    "m": ("fontSize": 1.35rem, "spacingBetweenIcons": 0.1875rem),
```
font-size is 1.375rem

Betrozov replied:
fixed

InnaAtanasova commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,119 @@
+    "l": ("fontSize": 2rem, "spacingBetweenIcons": 0.25rem)
+  );
+  $defaultSize: map-get($fd-rating-indicator-sizes, 'm');
+  $disabledOpacity: var(--sapContent_DisabledOpacity);
```
The idea of creating a variable is that it will be used in multiple places in the code and since this one is used only once you don't need to put the value in a variable

Betrozov replied:
fixed

InnaAtanasova commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,119 @@
+    "m": ("fontSize": 1.35rem, "spacingBetweenIcons": 0.1875rem),
+    "l": ("fontSize": 2rem, "spacingBetweenIcons": 0.25rem)
+  );
+  $defaultSize: map-get($fd-rating-indicator-sizes, 'm');
```
please use the naming convention adopted in the project. Should be something like `fd-rating-indicator-default-size`

Betrozov replied:
fixed

InnaAtanasova commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,119 @@
+  $defaultSize: map-get($fd-rating-indicator-sizes, 'm');
+  $disabledOpacity: var(--sapContent_DisabledOpacity);
+
+  %#{$block}-dynamic-text {
+    font-family: var(--sapFontFamily);
+    font-size: var(--sapFontSmallSize);
+    color: var(--sapContent_LabelColor);
+    margin-right: 0.5rem;
+  }
+
+  %#{$block}-input-hidden {
+    position: absolute;
+    opacity: 0.0001;
+  }
```
These placeholders are not used more than once, please move the code to the block directly

Betrozov replied:
fixed

InnaAtanasova commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,119 @@
+  $disabledOpacity: var(--sapContent_DisabledOpacity);
+
+  %#{$block}-dynamic-text {
+    font-family: var(--sapFontFamily);
```
font-family comes with the fd-reset function

Betrozov replied:
fixed

InnaAtanasova commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,119 @@
+$block: #{$fd-namespace}-rating-indicator;
+
+.#{$block} {
+  $ratedColor: var(--sapContent_RatedColor);
+  $unratedColor: var(--sapContent_UnratedColor);
+  $active-star-hover-opacity-transaction: opacity $fd-animation-speed ease-in;
```
Please use the naming convention of the project

Betrozov replied:
fixed

InnaAtanasova commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,119 @@
+  > .dynamic-text {
+    @extend %#{$block}-dynamic-text;
+
+    font-weight: normal;
```
it's probably coming from the reset mixin

Betrozov replied:
fixed

InnaAtanasova commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,119 @@
+    @extend %#{$block}-dynamic-text;
+
+    font-weight: normal;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
```
there's a `fd-ellipsis` mixin that you can use here

Betrozov replied:
fixed

InnaAtanasova commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,119 @@
+  @include fd-flex-center();
+
+  &__input {
+    @extend %#{$block}-input-hidden;
```
missing reset

Betrozov replied:
fixed

InnaAtanasova commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,119 @@
+  }
+
+  &__label {
+    @include action-cursor();
```
missing reset

Betrozov replied:
fixed, thank you for the comments

meganaconley commented:
I think @InnaAtanasova mentioned this issue, but there's more issues with the focus ring:

1. It doesn't show up at all on Mac using Chrome
2. When VoiceOver is on, there is one focus ring in the center that doesn't move even as you move selections

<img width="303" alt="Screen Shot 2020-08-27 at 12 15 49 PM" src="https://user-images.githubusercontent.com/57502177/91485343-95bdc300-e85f-11ea-82c1-09dbaebaed1c.png">

3. There should be a descriptive label for the radio group as a whole. It can be screenreader only if a visible label is not in the design.

However otherwise the keyboard navigation and labelling  of the individual stars is good.

Betrozov commented:
> 1. Clicking on the 4th star selects the 5th one too
> 
> <img alt="Screen Shot 2020-08-27 at 11 26 37 AM" width="372" src="https://user-images.githubusercontent.com/39598672/91462516-40ca7e80-e858-11ea-97ab-ae57c94d2ee3.png">
> 
> 2. You are missing the focus outline
> 
> <img alt="Screen Shot 2020-08-27 at 11 26 12 AM" width="243" src="https://user-images.githubusercontent.com/39598672/91462553-4c1daa00-e858-11ea-87c0-03db9f631c76.png">
> 
> 3. Missing examples with non-interactive and display mode stars

1. Fixed
2. Fixed
3. Non-interactive is equal to Read Only according to Design RatingIndicator (Fiori 3)

Betrozov commented:
> I think @InnaAtanasova mentioned this issue, but there's more issues with the focus ring:
> 
> 1. It doesn't show up at all on Mac using Chrome
> 2. When VoiceOver is on, there is one focus ring in the center that doesn't move even as you move selections
> 
> <img alt="Screen Shot 2020-08-27 at 12 15 49 PM" width="303" src="https://user-images.githubusercontent.com/57502177/91485343-95bdc300-e85f-11ea-82c1-09dbaebaed1c.png">
> 
> 3. There should be a descriptive label for the radio group as a whole. It can be screenreader only if a visible label is not in the design.
> 
> However otherwise the keyboard navigation and labelling of the individual stars is good.

1. fixed
2. fixed
3. fixed
fixed, thank you for the comments

InnaAtanasova commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,146 @@
+$block: #{$fd-namespace}-rating-indicator;
+
+.#{$block} {
+  $unratedColor: var(--sapContent_UnratedColor);
```
I think you forgot this one :)

Betrozov replied:
it used in multiple places in the code

InnaAtanasova replied:
I was referring to the naming of the variable. Should be `$fd-rating-indicator-unrated-color`

InnaAtanasova commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,433 @@
+                <label class="fd-rating-indicator__label" for="rating-xs-5"></label>
+            </div>
+
+            <span class="dynamic-text">(2 of 5)</span>
```
You can't have naming like this. If a project using fundamentals have a class dynamic-text there gonna be a styling conflict. We prefix our classes with `fd-` to avoid this. If this element is part of the Rating indicator it should be `fd-rating-indicator__dynamic-text`. If it's a stand-alone element it's styling should be in another file

Betrozov replied:
fixed

InnaAtanasova commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,146 @@
+
+  @include fd-flex-center();
+
+  &.is-display-mode {
```
can't this be a modifier class? Don't have to agree with me :D Just asking for your opinion on this :)

Betrozov replied:
I drew attention to the name of the classes for states, I'm about `.is-active`, `.is-disabled`, `.is-focus`, `.is-readonly`.
And `.is-display-mode` it is state but only for rating indicator

Betrozov replied:
thank you for the comments 🙂

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,433 @@
+        storyDescription: `
+| **Size**    | **rem**    | **Modifier class**       |
+| :---------- | :--------- | -----------------------: |
+| XS          | 0.75 rem   | <code>--xs</code>        |
```
I realized recently that rem is typed right next to the numbers, for example: 0.75rem, 1rem, 1.375rem etc. I would advise to change these (I will be doing the same for the docs I've written so far).

jessmetivier replied:
Otherwise, great table! 😄

Betrozov replied:
How would you like to change it? 🙂

jessmetivier replied:
Just remove the spacing between the number and "rem" (example: 0.75rem, 1rem). Everything else is perfect!

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,433 @@
+    </div>
+`;
+
+customIconSizes.parameters = {
```
Since this is for all the sizes, I would rename this section "Sizes"

Betrozov replied:
fixed

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,433 @@
+
+States.parameters = {
+    docs: {
+        storyDescription: `These Rating indicator types indicate two states: read-only and disabled.
```
Questions about this one:

1. It looks like there are 4 states in the example, so this can be confusing when you say there are two states. Which ones are read-only and which ones are disabled?

2. For the disabled and read-only instructions, when you say for instance: "the disabled attribute needs to be added to the..." are you referring to this `aria-disabled="true"`? Or are there 3 ways to disable the rating indicator?

Suggestion: Add headers (###) for **Disabled** and **Read-only** above each of these paragraphs (instructions). It'll be easier for the user to scan.

Betrozov replied:
fixed

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,433 @@
+
+Rtl.parameters = {
+    docs: {
+        storyDescription: 'This rating indicator is displayed from right-to-left on the screen..'
```
"The rating indicator can be displayed from right to left on the screen."

Also, there's something weird happening in the code example.. the numbers are jumbled, and the :Rtl is kind of awkward there. Anyway to put RTL on top, and right aligned?

Betrozov replied:
fixed

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,433 @@
+    title: 'Components/Rating Indicator',
+    parameters: {
+        description: `
+The rating indicator can be used to rate content or to indicate.
```
"The rating indicator displays a group of icons or images (usually stars) that indicate a rating. It allows users to rate content on a numeric scale, typically from 1 (lowest) to 5 (highest)."

Betrozov replied:
changed

thank you for the comments 🙂

meganaconley commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,433 @@
+    <div class="example-container">
+        <span>Default:</span>
+        <div class="fd-rating-indicator fd-rating-indicator--hide-dynamic-text">
+            <div class="fd-rating-indicator__container" aria-label="Rating of this item is 2 out of 5">
```
The label for the group should be more along the lines of describing what is being rated (ie. 'Star Rating for X (out of y)' or just 'Star Rating (out of x)' if the context is going to be clear. The value for the item is already listed in text below.

Betrozov replied:
changed to `Star Rating (out of x)`

meganaconley commented:
There is still no focus outline in Chrome. When VoiceOver is on, the outline appears but is off-center.

<img width="162" alt="Screen Shot 2020-08-28 at 1 11 37 PM" src="https://user-images.githubusercontent.com/57502177/91611457-2ebc0f00-e930-11ea-8339-6ed5b67f064e.png">

Betrozov commented:
> There is still no focus outline in Chrome. When VoiceOver is on, the outline appears but is off-center.
> 
> <img alt="Screen Shot 2020-08-28 at 1 11 37 PM" width="162" src="https://user-images.githubusercontent.com/57502177/91611457-2ebc0f00-e930-11ea-8339-6ed5b67f064e.png">

fixed
thank you for the comments 🙂

droshev commented:
@Betrozov can you handle the theming specific styling in https://github.com/SAP/fundamental-styles/tree/master/src/theming
you can see an example: https://github.com/SAP/fundamental-styles/pull/1423/files

JKMarkowski commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,190 @@
+  position: relative;
+
+  &__container {
+    @include fd-fiori-focus-within();
```
I know `focus-within` is great and it fits well for this case, but it's not supported by IE11. You can achieve same thing by having pseudo element on label, when input is focused. So `.fd-input:focus + .fd-label::after { position: absolute ...;  } with position relative on container element

Betrozov replied:
extended documentation
thank you for the comments 🙂

JKMarkowski reviewed (commented):
There is one thing from me regarding focus and IE11

Betrozov commented:
> @Betrozov can you handle the theming specific styling in https://github.com/SAP/fundamental-styles/tree/master/src/theming
> you can see an example: https://github.com/SAP/fundamental-styles/pull/1423/files

fixed
thank you for the comments 🙂

droshev commented on src/theming/sap_fiori_3.scss:
```diff
@@ -1,3 +1,19 @@
+
+  /* Rating Indicator */
+  // default mode
+  --sapRating_Indicator_Selected_Icon_Color: #d08014;
```
instead of hardcoding the values, use the parameter name `--sapContent_RatedColor`

Betrozov replied:
fixed

droshev commented on src/theming/sap_fiori_3.scss:
```diff
@@ -1,3 +1,19 @@
+  // default mode
+  --sapRating_Indicator_Selected_Icon_Color: #d08014;
+  --sapRating_Indicator_Selected_Hover_Opacity: 0.9;
+  --sapRating_Indicator_Unselected_Icon_Color: #89919a;
```
`--sapContent_UnratedColor

`

Betrozov replied:
fixed

droshev commented on src/theming/sap_fiori_3.scss:
```diff
@@ -1,3 +1,19 @@
+  --sapRating_Indicator_Selected_Hover_Opacity: 0.9;
+  --sapRating_Indicator_Unselected_Icon_Color: #89919a;
+  // Readonly mode
+  --sapRating_Indicator_Readonly_Selected_Color: #d08014;
```
Can you use the CSS variables instead of hardcoding the values?
for the colors bellow

Betrozov replied:
fixed

InnaAtanasova commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,206 @@
+    }
+  }
+
+  &__with-half-star {
```
following BEM this should be a modifier class: `fd-rating-indicator fd-rating-indicator--half-star`
`__` is for element
`--` is for modifier

Betrozov replied:
fixed

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,510 @@
+DisplayMode.parameters = {
+    docs: {
+        storyDescription: `
+To make the \`Rating Indicator\` component display mode needs to be added class \`.is-display-mode\` to the \`fd-rating-indicator\` element.
```
"Data can be presented as label-value field pairs without editable fields in display-only forms, see **Form Grid** for more details. If you want to include a rating indicator in a display-only form, add the `.is-display-mode` to the `fd-rating-indicator` element."

jessmetivier replied:
It would be great to link **Form Grid** to its corresponding page.

Betrozov replied:
fixed

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,510 @@
+Focused.parameters = {
+    docs: {
+        storyDescription: `
+By default there is built-in focus indicator for \`Rating Indicator\` component, which is not supported by IE11.
```
"By default, rating indicator is focusable. However, it is not supported by IE11. To display a focusable rating indicator on IE11 _(browsers?)_, add the `.is-focus` class to the component."

jessmetivier replied:
Not sure if it's just IE11 browsers, or applications? I'll let you be the judge of what should go there.

Betrozov replied:
I think we can just use IE11 here without prefixes browsers or applications

jessmetivier replied:
Okay sounds good.

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,510 @@
+    }
+};
+
+export const RatingIndicatorWithDifferentMaxValue = () => `
```
I would title this as "Different values" or "Maximum values"

And for all the titles, there should only be a capitalization on the first letter of the first word i.e. "**D**isplay mode"

Description:
"It is possible to display the rating indicator with a different number of values. It is highly recommended to use 5 as the maximum value, however you can go up to 7 if it is ideal for your use case."

Betrozov replied:
fixed

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,510 @@
+    </div>
+`;
+
+Rtl.parameters = {
```
Capitalize RTL always

Betrozov replied:
fixed

Betrozov commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,510 @@
+ReadOnly.parameters = {
+    docs: {
+        storyDescription: `
+To make the \`Rating Indicator\` component read-only needs to be added attribute \`[readonly]\` 
```
We have 3 different way to define this state

Betrozov commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,510 @@
+Disabled.parameters = {
+    docs: {
+        storyDescription: `
+To disable the \`Rating Indicator\` component needs to be added attribute \`[disable]\` 
```
the same answer as for `readonly `

Betrozov commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,510 @@
+</div>
+`;
+
+HalfSizes.parameters = {
```
fixed

Betrozov commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,510 @@
+        description: `
+The rating indicator displays a group of icons or images (usually stars) that indicate a rating.
+It allows users to rate content on a numeric scale, typically from 1 (lowest) to 5 (highest).
+Although the maximum amount is 7 for the icons or images, it is highly recommended to display the default amount of 5."
```
😄

mikerodonnell89 commented:
Leftmost star extends ever so slightly past the focus outline

![Screen Shot 2020-08-31 at 3 34 33 PM](https://user-images.githubusercontent.com/2471874/91771096-9af07a00-eb9f-11ea-8bc9-55e621724663.png)

meganaconley commented:
On the focus ring issue... I'm sorry if I wasn't clear, but the focus ring needs to be around the individual star as the user moves their focus from one to another, not the entire set of options. 

It now looks better in VoiceOver (VO is placing that thick square outline) but still a bit off center. The focus ring should appear on each star kind of like this even when VO is not on:

<img width="336" alt="Screen Shot 2020-08-31 at 2 51 15 PM" src="https://user-images.githubusercontent.com/57502177/91772393-a80a6a80-eb99-11ea-8e4f-105f35bd3850.png">

Betrozov commented:
> On the focus ring issue... I'm sorry if I wasn't clear, but the focus ring needs to be around the individual star as the user moves their focus from one to another, not the entire set of options.
> 
> It now looks better in VoiceOver (VO is placing that thick square outline) but still a bit off center. The focus ring should appear on each star kind of like this even when VO is not on:
> 
> <img alt="Screen Shot 2020-08-31 at 2 51 15 PM" width="336" src="https://user-images.githubusercontent.com/57502177/91772393-a80a6a80-eb99-11ea-8e4f-105f35bd3850.png">

@meganaconley fixed plz check it one more time

JKMarkowski commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,220 @@
+    &:checked {
+      @include fd-fiori-focus(0) {
+        + .#{$block}__label {
+          outline-offset: 0;
+          outline-width: var(--sapContent_FocusWidth);
+          outline-color: var(--sapContent_FocusColor);
+          outline-style: var(--sapContent_FocusStyle);
```
Let's change it to pseudo element `&::after { position: absolute; top: 0; left: 0; width: 100%; height: 100%; border: sapContent_FocusStyle sapContent_FocusColor sapContent_FocusWidth}` 
With `.fd-rating-indicator__container { position: relative; }`
![image](https://user-images.githubusercontent.com/26483208/91825497-a8a80d00-ec3c-11ea-8817-2fc5f804b6fd.png)

JKMarkowski commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,220 @@
+
+    display: flex;
+    align-items: center;
+    float: left;
```
What about adding `display: flex` to `fd-rating-indicator__container`, to get rid of floats?

JKMarkowski replied:
Impossible due to accessibiliyt stuff

JKMarkowski commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,220 @@
+    }
+
+    @include fd-rtl() {
+      float: right;
```
With flex on container we would be able to remove it

JKMarkowski replied:
Impossible due to accessibiliyt stuff

JKMarkowski reviewed (changes requested):
Hello @Betrozov I added another comments, let me know if something is not clear

JKMarkowski reviewed (approved):
Looks good to me now. Good Job!

JKMarkowski commented on src/rating-indicator.scss:
```diff
@@ -0,0 +1,220 @@
+      }
+
+      .#{$block}__label {
+        margin-right: $fd-rating-indicator-sizes-spacing-between-icons;
```
Margin doesn't support RTL 
![image](https://user-images.githubusercontent.com/26483208/91840491-0e9e8f80-ec51-11ea-8b08-521bdb048d23.png)
You can change it to `margin-left`, or switch `&:last-child` with `&:first-child` on RTL

Betrozov replied:
fixed

JKMarkowski reviewed (changes requested):
Just one more thing. The rest LGTM

JKMarkowski reviewed (approved):
LGTM

mikerodonnell89 commented:
The Fiori 3 specs state that the ratings indicator can only display half stars in display/read-only mode

Betrozov commented:
> The Fiori 3 specs state that the ratings indicator can only display half stars in display/read-only mode

https://github.com/SAP/fundamental-styles/pull/1465#pullrequestreview-477937919

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,529 @@
+CustomIcons.parameters = {
+    docs: {
+        storyDescription: `
+To make the rating indicator to use custom icons needs to be added class \`.fd-rating-indicator--icon\` 
```
I'm not seeing how this is working in the code, perhaps it's more complex than this explanation?

Betrozov replied:
in the code updated this line
```html
<div class="fd-rating-indicator fd-rating-indicator--icon"
    style="--sapRating-indicator-icon-rated: url(${iconRated}); --sapRating-indicator-icon-unrated: url(${iconUnrated})">
```
if we go to the demo and look at the Custom icons section you will see smiles icons instead of star icon

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,529 @@
+</div>
+`;
+
+ReadOnly.parameters = {
```
Change name to "Read-Only" for consistency.

Betrozov replied:
fixed

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,529 @@
+        storyDescription: `
+To disable rating indicator, the \`[disable]\` attribute needs to be added to all inputs. 
+Additionally, one of the selectors listed below needs to be added to the \`fd-rating-indicator\` element:
+\`.is-disable\`
```
Add bullet points here

Betrozov replied:
added

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,529 @@
+    </div>
+`;
+
+RTL.parameters = {
```
Still displaying as "Rtl" better to capitalize all letters

Betrozov replied:
fixed

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,529 @@
+        storyDescription: `
+To make the rating indicator read-only, the \`[readonly]\` attribute needs to be added to all inputs. 
+Additionally, one of the selectors listed below needs to be added to the \`fd-rating-indicator\` element:
+\`.is-readonly\`
```
Add bullet points here

Betrozov commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,529 @@
+HalfSizes.parameters = {
+    docs: {
+        storyDescription: `
+To make the rating indicator to use half star needs to be added class \`.fd-rating-indicator--half-star\` 
```
fixed

bcullman commented:
The example for read-only is interactive, but that seems wrong. 

![ratings](https://user-images.githubusercontent.com/5056972/92074667-7e08a200-ed6b-11ea-8187-a9a1b5d70a12.gif)

Betrozov commented:
> The example for read-only is interactive, but that seems wrong.
> 
> ![ratings](https://user-images.githubusercontent.com/5056972/92074667-7e08a200-ed6b-11ea-8187-a9a1b5d70a12.gif)

Hi @bcullman. 
Thank you for the comment
What browser and OS did you use to find this issue?

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,537 @@
+Sizes.parameters = {
+    docs: {
+        storyDescription: `
+| **Size**    | **rem**    | **Modifier class**       |
```
Please remove the space between the number and "rem"
0.75rem
1rem
1.375rem
2rem
etc.

Betrozov replied:
fixed

jessmetivier commented on stories/rating-indicator/rating-indicator.stories.js:
```diff
@@ -0,0 +1,537 @@
+RTL.storyName = 'RTL';
+RTL.parameters = {
+    docs: {
+        storyDescription: 'The rating indicator can be displayed from right to left on the screen.'
```
Please add this to the text:

The rating indicator can be displayed from right to left on the screen so that it may be used internationally.

Betrozov replied:
fixed

jessmetivier reviewed (changes requested):
Hi,
Everything looks great!
Please make minor adjustments and ping me on Slack for approval ;)

mikerodonnell89 commented:
There also needs to be an option for custom hover icon, according to the rating indicator spec.  Would we need half-ratings for the custom icons as well?  I don't see this mentioned in the specs but I'm assuming the answer is yes.

mikerodonnell89 reviewed (commented):
There also needs to be an option for custom hover icon, according to the rating indicator spec.  Would we need half-ratings for the custom icons as well?  I don't see this mentioned in the specs but I'm assuming the answer is yes.

jessmetivier reviewed (approved):
Great work!

bcullman commented:
@Betrozov 

chrome mac:

![image](https://user-images.githubusercontent.com/5056972/92772502-a5a1d080-f350-11ea-988b-1e74cd81f5aa.png)

Betrozov commented:
> @Betrozov
> 
> chrome mac:
> 
> ![image](https://user-images.githubusercontent.com/5056972/92772502-a5a1d080-f350-11ea-988b-1e74cd81f5aa.png)

fixed
thank you for the comment

bcullman commented:
@Betrozov not sure it is fixed. 

The text now says disabled. disabled and read only are different states, handled by screen readers differently.

Also, the element is now not focusable. Again, this is probably because you made it disabled.

stefanoScalzo commented:
the control margin isnt adding any margin

Betrozov commented:
> the control margin isnt adding any margin

control margin works only for cozy and compact

Betrozov commented:
> @Betrozov not sure it is fixed.
> 
> The text now says disabled. disabled and read only are different states, handled by screen readers differently.
> 
> Also, the element is now not focusable. Again, this is probably because you made it disabled.

@bcullman I rolled back `disabled` to `readonly` to make
one notes input radio does not support readonly attribute https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly

jacobdevera commented:
According to [the spec](https://experience.sap.com/fiori-design-web/rating-indicator/#types), there should only be an 'interactive' and 'disabled' state, as the 'non-interactive' state (presumably read-only) is not yet available, so it appropriate to omit read-only for now.
